test/e2e/apps/daemon_set.go

```diff
@@ -746,3 +780,189 @@ func waitFailedDaemonPodDeleted(c clientset.Interface, pod *v1.Pod) func() (bool
 		return false, nil
 	}
 }
+
+// deleteDaemonSetAndOrphanPods deletes the given DaemonSet and orphans all its dependents.
+// It also checks that all dependents are orphaned, and the DaemonSet is deleted.
+func deleteDaemonSetAndOrphanPods(
```

Comment:
`deleteDaemonSetAndOrphan`. DaemonSet doesn't just own pods, it owns controllerrevisions too.